app/models/email_alert_signup.rb

```diff
@@ -0,0 +1,54 @@
+require 'active_model'
+
+class EmailAlertSignup
+  include ActiveModel::Model
+
+  validates_presence_of :content_item
+
+  delegate :title, :description, to: :content_item
+  delegate :tags, to: :"content_item.details"
+
+  attr_reader :subscription_url
+
+  def initialize(content_item)
+    @content_item = content_item
+  end
+
+  def save
+    if valid?
+      @subscription_url = find_or_create_subscription.subscriber_list.subscription_url
+      true
+    else
+      false
+    end
+  end
+
+  def find_or_create_subscription
+    EmailAlertFrontend.services(:email_alert_api)
+      .find_or_create_subscriber_list(subscription_params)
+  end
+
+  def breadcrumbs
+    return {} if raw_breadcrumbs.blank?
+
+    raw_breadcrumbs.reverse.reduce { |memo, crumb|
+      crumb.merge(parent: memo)
+    }
+  end
+
+private
+  attr_reader :content_item
+
+  def subscription_params
+    {
+      title: title,
+      tags: tags.marshal_dump
```

Comment:
Would be good to have a comment or something to indicate why `marshal_dump`. Or better yet, [extract method](http://refactoring.com/catalog/extractMethod.html)!

```
...
  tags: convert_openstruct_to_hash(tags)
...

def convert_openstruct_to_hash(open_struct)
  open_struct.marshal_dump
end

```